packages/ui/src/ui-component/dropdown/AsyncDropdown.js

```diff
@@ -105,7 +105,7 @@ export const AsyncDropdown = ({
         })()
 
         // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, [])
+    }, [credentialNames])
```

Comment:
this causes problem as well.

Just put in a custom tool or assistant:

1.) 
<img width="491" alt="image" src="https://github.com/FlowiseAI/Flowise/assets/26460777/beb086b8-4165-4fa9-9511-d427351b55a5">

You will see it constantly loading

2.) From the terminal, you can see the method gets called infinitely:
<img width="710" alt="image" src="https://github.com/FlowiseAI/Flowise/assets/26460777/f8ef1b6a-c629-43de-ad10-1ea7ceae1f3d">

Reply:
resolved